engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/AncientDataMotionStrategy.java

```diff
@@ -392,15 +405,16 @@ protected Answer copyVolumeBetweenPools(DataObject srcData, DataObject destData)
                     return answer;
                 }
             } catch (Exception e) {
-                if (imageStore.exists(objOnImageStore)) {
+                if (objOnImageStore != null) {
```

Comment:
is not null enough? what if it is an object but doesn't exist? or it does but not on `imageStore`?

Reply:
imagestore.exists() eventually calls volumeDataStoreDao.findByStoreVolume(...) for volumes, which in case of multiple entries could possibly return the wrong record

Reply:
ok, but still, the `objOnImageStore` may be a valid object that represents nothing on our `imageStore`, can it? shouldn't we protect against that as well?

Reply:
@DaanHoogland I didn't quite understand your previous comment. Could you please explain what you mean by it. Thanks

Reply:
@Pearl1594 what you are saying might be true but during catch() we have to check if DB entry really exists or not and process the image storage object by marking it OperationFailed. objOnImageStore will always be != null (_"DataObject objOnImageStore = imageStore.create(srcData);"_) since no where in try{} block we are setting to null or changing the object.

@DaanHoogland in that method we are dealing only with database entries. If you are asking if it represents nothing on imagestore "physically" then the answer is we don't need to have concern about that.